### PR TITLE
Tests: Assure that nobody breaks UserspaceEmulator accidentally

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
**Tests: Assure that UserspaceEmulator via ue ls**

**Tests: Rename write-oob**
The "ue-" prefix makes it clearer that it should be run with the
UserspaceEmulator

**Tests: Change test-filtering mechanism**
We have a new config argument to add space separated exclude regex'
This is separate from "NotTestsPattern", because these are still Tests,
although they are not supposed to be run by the runner

This also adds the test for a working UserspaceEmulator to the tests run

Oh yeah you might need to clean the disk image after this commit, small heads up
And yes calling system isn't the best way of doing it, I was just unable to get execve working

### CODE REVIEW

File: Userland/Utilities/run-tests.cpp
Comment (ADKaster) [inline Userland/Utilities/run-tests.cpp:328]: This feels kinda weird, now there's three ways to mark a test as "skip"

SkipTests takes a space separated list of tests to skip

SkipDirectories takes a space separated list of directories to skip

And

SkipRegex takes a path regex to.. ignore? But your comment on the PR says these are different from ignore? You're adding them to the ignore regex 🤔. Not the skip regex. What's the goal here again? 😅

Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp:328]: The difference was supposed to be the name/intention, but good catch, I will change that

Comment (ADKaster) [reply Userland/Utilities/run-tests.cpp:328]: The phrasing of "ignore" vs "skip" seems obvious in hindsight, idk why I didn't think of it the first go around. If you wanted to change NotTestPattern to IgnoreRegex that would probably be a lot clearer.

Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp:328]: Well it should still skip the ue specific test,
and while trying to implement it I somehow broke something with the regex byte-code engine, so will look into it tomorrow

File: Base/home/anon/.config/Tests.ini
Comment (alimpfard) [inline Base/home/anon/.config/Tests.ini:5]: This pattern...won't match anything...
$ is the end-of-line assertion, nothing can match after that

Comment (sin-ack) [reply Base/home/anon/.config/Tests.ini:5]: Seems like `^` and `$` are flipped. :^)

File: Tests/UserspaceEmulator/test-run-ls.cpp
Comment (sin-ack) [inline Tests/UserspaceEmulator/test-run-ls.cpp:1]: Typo in commit title, something like ``Tests: Assure that UserspaceEmulator does not regress via `ue ls` ``?

File: Base/home/anon/.config/Tests.ini
Comment (Hendiadyoin1) [reply Base/home/anon/.config/Tests.ini]: flipped those now

File: Tests/UserspaceEmulator/test-run-ls.cpp
Comment (Hendiadyoin1) [reply Tests/UserspaceEmulator/test-run-ls.cpp:1]: I needed to abbreviate UserspaceEmulator, but yes verbs might be a good idea

File: Userland/Utilities/run-tests.cpp
Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp]: This should be fixed now

File: AK/Unaligned.h
Comment (ADKaster) [inline AK/Unaligned.h:13]: The read part of this is already in AK in ByteReader.h. could we get some reuse going?

Comment (Hendiadyoin1) [reply AK/Unaligned.h]: Ah then I'll just use that then

File: Userland/Utilities/run-tests.cpp
Comment (alimpfard) [inline Userland/Utilities/run-tests.cpp:331]: Doesn't look like these braces are necessary?

File: Userland/Utilities/run-tests.cpp
Comment (alimpfard) [inline Userland/Utilities/run-tests.cpp:343]: This seems a little odd, splitting regular expressions with space could make them invalid, for instance: `SomeRandomTest-[^ ]+` would be transformed to `SomeRandomTest-[^|]+` with this.

File: Userland/Utilities/run-tests.cpp
Comment (alimpfard) [inline Userland/Utilities/run-tests.cpp:345]: Maybe say what this is about in the log? it could also be put behind a debug flag if there is one

**Review by alimpfard — COMMENTED** [review]
Looks generally alright, just a few things

File: Userland/Utilities/run-tests.cpp
Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp:345]: I thought I removed that one...

File: Userland/Utilities/run-tests.cpp
Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp:343]: So I just take a patten instead

File: Userland/Utilities/run-tests.cpp
Comment (Hendiadyoin1) [reply Userland/Utilities/run-tests.cpp:331]: removed

**Review by alimpfard — APPROVED** [review]
LGTM.
Let's just wait for CI before merging.

**Comment by Hendiadyoin1** [comment]
Wooo only a new 163.857064s test!

**Comment by alimpfard** [comment]
Maybe that'll motivate someone to work on ue's perf some day

**Pull request merged**
